src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -51,12 +55,22 @@ class SpfxProjectUpgradeCommand extends Command {
     '1.8.0'
   ];
 
+  // https://github.com/SharePoint/sp-dev-docs/wiki/SharePoint-Framework-v1.8-release-notes
+  // [...] version 0.5.7 of the rush-stack-compiler-2.7 package works, version 0.6.8 of the rush-stack-compiler-2.9 package works, 0.5.9 of the rush-stack-compiler-3.0 package works, and 0.1.6 of the rush-stack-compiler-3.3 package works.
+  private supportedTypeScriptVersions: { tscVersion: string, rushPackageVersion: string }[] = [
```

Comment:
I wonder to what extent the list of supported TypeScript versions should be linked to specific SharePoint Framework versions. I can imagine, that over time new TypeScript versions will be supported in SPFx but only in combination with newer versions of SPFx. So if we were to centralize this list in _project-upgrade.ts_ it would over time become cluttered with the references to the different versions of SPFx. Perhaps it wold be a better idea to link the list of supported versions to the specific SPFx version in _upgrade-{version}.ts_. The idea is to keep _project-upgrade.ts_ as generic as possible and keep all version-specific logic isolated.